packages/core/src/common/utils.ts

```diff
@@ -22,6 +22,14 @@ export function isFunction(value: any): value is Function {
     return typeof value === "function";
 }
 
+export interface IHasName {
```

Comment:
what's the use case for this? if we keep it, it should be called `INamed`

Reply:
I'm adapting https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging . The `.name` is referencing `Function.name` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) which evaluates to the class name for constructors which is what we want 90% of the time, but for some reason Typescript types don't include `.name` as a field in `Function`. This was the easiest way to express the type.

Reply:
cool, makes sense, can you leave a comment explaining that?